### DISCUSSION TITLE
Wildcard for H2O_WAVE_ALLOWED_ORIGINS

### DISCUSSION THREAD

**Discussion by maxjeblick** [h2oai/wave#2354] [Q&A]
Hi!

Is it possible to set `H2O_WAVE_ALLOWED_ORIGINS` to match an arbitrary url?

Background:
I'm using LLM Studio with port forwarding on a cloud instance. The dynamically generated link to access localhost:10101 varies for each run. Currently, I need to
- Start the instance
- Manually copy the link that forwards 10101
- Log into the cloud instance
- Set export `H2O_WAVE_ALLOWED_ORIGINS="https://..."`  (note that the link needs to be without / at the end, this took me a while to figure out. Maybe trailing / can be removed automatically?)
- Start wave

I'm looking for a solution that would allow me to write a dockerfile that automatically starts llm studio, so that I can directly open the link.

Sidenote:
Jupyter uses  `--NotebookApp.allow_origin='*'` see [here](https://jupyter-notebook.readthedocs.io/en/4.x/config.html). I tried `*`, but apparently this is not working within wave.

Thanks,
Max

**Comment by mturoci** [comment]
Hi @maxjeblick, I am currently on parental leave so sorry for taking so long. I think we can support `*`, feel free to file a feature request.

**Reply by maxjeblick** [reply]
Enjoy your parental leave! I'll open an issue.

**Reply by maxjeblick** [reply]
https://github.com/h2oai/wave/issues/2370
